### ISSUE TITLE
Wayland Window Cursor Lag

### ISSUE THREAD

**Issue by Trynera** [Avdan-OS/Compositor#75]
### Please make sure your system is up to date

- [X] Yes, my system is up to date

### Select your CPU architecture

x86-64

### Select your Linux distribution

Archlinux

### Please enter your RAM size

8 GB

### Please enter your GPU model

VirtualBox GPU ig

### Please explain what went wrong

The Mouse Cursor starts lagging once inside the Terminal Window! I've put a Video to explain what I mean. It could be the Terminal but probably is the Compositor (as just being inside the top bar lags the cursor)

### Provide us links that may help us fix this issue, if you have any

https://user-images.githubusercontent.com/96610933/230912869-7886c191-596a-4efd-8f94-65babca004ee.mp4

### Provide a screenshot of the problem, if you can

How tf am I suppost to make a ss of this tf?

### Additional information

Tell everyone I'm cool ig xd

### Please make sure this issue is not a duplicate of an existing issue.

- [X] Yes, I have searched for this issue on this repository and none were found

**Comment by Trynera** [comment]
basically, the cursor goes 60 fps outside the window, and what feels like 15 inside the window

**Comment by Trynera** [comment]
I've also realised that the Terminal Window does some checks once the Cursor is inside the window! Another Video showing what I mean

https://user-images.githubusercontent.com/96610933/230915970-17a0d63b-a955-47ad-9b96-5c6802740c31.mp4

**Comment by Sammy99jsp** [comment]
Interesting; I wonder if this could be recreated on real hardware.

Possibly to do with the graphics driver?

**Comment by Trynera** [comment]
I don't think it could be graphics drivers as the cursor alone works perfectly fine, until it's inside the window. Outside the compositor it also worked perfectly fine even inside a window. Maybe it's doing unnessecery computations when inside a window?
